pkg/admission/admission_controller_test.go

```diff
@@ -568,7 +568,9 @@ func TestMutate(t *testing.T) {
 	assert.Check(t, resp.Allowed, "response not allowed for unknown object type")
 	assert.Check(t, len(resp.Patch) > 0, "empty patch for deployment")
 	annotations := annotationsFromDeployment(t, resp.Patch)
-	assert.Equal(t, annotations[common.UserInfoAnnotation].(string), "{\"user\":\"testExtUser\"}")
+	annotationValue, ok := annotations[common.UserInfoAnnotation].(string)
+	assert.Check(t, ok, "UserInfoAnnotation value is not a string")
```

Comment:
nit: it's more reasonable to just bail out with `assert.Assert()` immediately. It makes no sense to continue the test in such cases (also applies to lines 844, 856, 868).